CMakeLists.txt

```
@@ -117,91 +117,95 @@ option(VITA3K_FORCE_SYSTEM_BOOST "Force Vita3K build process to use the Boost ve
+
+		      # If no build of the custom distribution is found, compile it
+		      if(NOT Boost_FOUND)
+		          message(STATUS "No existing custom distribution of Boost has been found in ${BOOST_INSTALLDIR}. Proceeding to compile Boost...")
+
+		          # Set compilation toolset for b2 and Boost builds
+		          if (CMAKE_HOST_SYSTEM_NAME STREQUAL "Windows")
+		              if(CMAKE_CXX_COMPILER_ID STREQUAL "Clang")
+		                  if(CMAKE_GENERATOR_TOOLSET STREQUAL "ClangCL")
+		                      # Proper toolset should be "clang-win", but b2 has a bug that makes it spam Visual Studio instances if set
+		                      # Since Clang for Windows is ABI-compatible with MSVC, using MSVC is fine
+		                      set(BOOST_TOOLSET "msvc")
+		                  else()
+		                      message(WARNING "Boost compilation on Windows requires the use of a clang-cl compiler running inside a Visual Studio Developer CLI environment with a Windows C++ SDK available.")
+		                      set(BOOST_TOOLSET "msvc")
+		                  endif()
+		              else()
+		                  set(BOOST_TOOLSET "msvc")
+		              endif()
+		          elseif(CMAKE_HOST_SYSTEM_NAME STREQUAL "Linux")
+		              if(CMAKE_CXX_COMPILER_ID STREQUAL "Clang")
+		                  set(BOOST_TOOLSET "clang")
+		              elseif(CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
+		                  set(BOOST_TOOLSET "gcc")
+		              endif()
+		          endif()
+
+		          # Determine b2 executable name depending on host system
+		          if(CMAKE_HOST_SYSTEM_NAME STREQUAL "Windows")
+		              set(b2_EXECUTABLE_NAME "b2.exe")
+		          else()
+		              set(b2_EXECUTABLE_NAME "b2")
+		          endif()
+
+		          # Find b2
+		          find_program(b2 NAME ${b2_EXECUTABLE_NAME} PATHS ${BOOST_SOURCEDIR} NO_CACHE NO_DEFAULT_PATH)
+
+		          # Compile b2 if it isn't found
+		          if(b2 STREQUAL "b2-NOTFOUND")
+		              message(STATUS "The b2 buildsystem executable hasn't been found in your system. Compiling b2...")
+		              b2_build()
+
+		              # Find b2 again after compilation
+		              find_program(b2 NAME ${b2_EXECUTABLE_NAME} PATHS ${BOOST_SOURCEDIR} NO_CACHE NO_DEFAULT_PATH)
+		          endif()
+
+		          # Compile Boost
+		          message(STATUS "Compiling Boost...")
+		          boost_compile()
+
+		          # Find Boost again
+		          find_package(Boost COMPONENTS ${BOOST_MODULES_TO_FIND} PATHS ${BOOST_INSTALLDIR} NO_DEFAULT_PATH)
+		      else()
+		          message(STATUS "Custom Boost distribution found in ${BOOST_INSTALLDIR}")
+		      endif()
+		  endif()
+	else()
+		  # Try to find Boost on the system and CMake's default paths
+		  set(Boost_USE_STATIC_LIBS ON)
+		  find_package(Boost COMPONENTS ${BOOST_MODULES_TO_FIND} QUIET)
```

```suggestion
		          find_package(Boost COMPONENTS ${BOOST_MODULES_TO_FIND} PATHS ${BOOST_INSTALLDIR} NO_DEFAULT_PATH REQUIRED)
		      else()
		          message(STATUS "Custom Boost distribution found in ${BOOST_INSTALLDIR}")
		      endif()
		  endif()
	else()
		  # Try to find Boost on the system and CMake's default paths
		  set(Boost_USE_STATIC_LIBS ON)
		  find_package(Boost COMPONENTS ${BOOST_MODULES_TO_FIND} REQUIRED)
```
Boost is required to build Vita3K. might as well fail as soon as possible if it is not found.

---

Done.